eras/conway/impl/src/Cardano/Ledger/Conway/Rules/Gov.hs

```diff
@@ -232,17 +258,32 @@ govTransition = do
                   Set.filter ((/= expectedNetworkId) . getRwdNetwork) $ Map.keysSet wdrls
              in Set.null mismatchedAccounts
                   ?! TreasuryWithdrawalsNetworkIdMismatch mismatchedAccounts expectedNetworkId
-          NewCommittee _mPrevGovActionId _oldColdCreds newCommittee ->
-            let minCommitteeSize = pp ^. ppMinCommitteeSizeL
-                newCommitteeSize = fromIntegral . Map.size $ newCommittee ^. committeeMembersL
-             in newCommitteeSize
-                  >= minCommitteeSize
-                    ?! NewCommitteeSizeTooSmall newCommitteeSize minCommitteeSize
+          UpdateCommittee _mPrevGovActionId membersToRemove membersToAdd qrm -> do
+            _ <- checkCommitteeSize
+            _ <- checkUnexpectedMembers
+            checkExpirationEpoch
+            where
+              checkCommitteeSize =
+                let minCommitteeSize = pp ^. ppMinCommitteeSizeL
+                    newCommittee = updatedCommittee membersToRemove membersToAdd qrm committee
+                    committeeSize = fromIntegral . Map.size $ foldMap' (^. committeeMembersL) newCommittee
+                 in committeeSize
+                      >= minCommitteeSize
+                        ?! NewCommitteeSizeTooSmall committeeSize minCommitteeSize
+              checkUnexpectedMembers =
+                let currentMembers = Map.keysSet $ foldMap' (^. committeeMembersL) committee
+                    unexpectedMembers = membersToRemove `Set.difference` currentMembers
+                 in Set.null unexpectedMembers ?! UnexpectedCommitteeMembers unexpectedMembers
+              checkExpirationEpoch =
+                let maxTermLimit = pp ^. ppCommitteeTermLimitL
+                    invalidMembers =
+                      Map.filter (> EpochNo (fromIntegral maxTermLimit) + currentEpoch) $ membersToAdd
```

Comment:
I agree that it's a little awkward that it's an absolute epoch instead of a number of epochs, but it is that way for a good reason: you want full predictability here. For example, if there's going to be a schedule where one third of the CC is replaced each year it's much better to align the schedule exactly, instead of having a few weeks where some time out but others don't.

Two notes about `ccTermLimit`:
- First, in my PR to the CIP I've renamed it to `ccMaxTermLength`, because that's what it actually is.
- My plan is to enforce it in `ENACT`. The reason is that this check can become successful over time, and we only know when it'll be successful in `ENACT`.

Reply:
> if there's going to be a schedule where one third of the CC is replaced each year it's much better to align the schedule exactly

Would it be as easy as submitting an `UpdateCommittee` GovAction that only updates the lifetime of 1/3 of the committee to the same duration?